oximeter/collector/src/results_sink.rs

```diff
@@ -98,6 +99,41 @@ pub async fn database_inserter(
                     );
                 }
             }
+
+            // Our internal testing rack will be running a ClickHouse cluster
+            // alongside a single-node installation for a while. We want to handle
+            // the case of these two installations running alongside each other, and
+            // oximeter writing to both of them. On our production racks ClickHouse
+            // will only be run on single-node modality, so we want to ignore all
+            // cases where the `ClickhouseClusterNative` service is not available.
+            if let Some(cluster_client) = &cluster_client {
+                match cluster_client.ping().await {
+                    Ok(()) => {
+                        debug!(
+                            log,
+                            "inserting {} samples into cluster database",
+                            batch.len();
+                        );
+                        match cluster_client.insert_samples(&batch).await {
+                            Ok(()) => trace!(
+                                log,
+                                "successfully inserted samples into cluster";
+                            ),
+                            Err(e) => {
+                                warn!(
+                                    log,
+                                    "failed to insert some results into metric cluster DB: {e}"
+                                );
+                            }
+                        }
+                    }
+                    Err(e) => info!(
+                        log,
+                        "ClickHouse cluster native connection unavailable: {e}"
```

Comment:
If we do keep this, could we print the error as with `InlineErrorChain` or similar? Squashing it to a string loses a lot of details usually.

Reply:
Done in [08b9bb4](https://github.com/oxidecomputer/omicron/pull/7565/commits/08b9bb4f40efedccb18e7c06173557859d5339f1)